### PR TITLE
Add stuff for native hooks in patching.md

### PR DESCRIPTION
**Pull request by Onixiya**
Would most def need someone else more experienced then me to look it over but i think its a good start

### CODE REVIEW

**Review by loukylor — CHANGES_REQUESTED** [review]
I'm not sure about the actual content, but PLEASE add any amount of whitespace, and limit the length of your lines. As it stands right now, your code is too hard to read. You should add comments that refer to content in the paragraph below too. Also, utilize backticks (\`\`) for your code snippets (`_originalMethod`, `Il2CPP.PointerToValueGeneric`, etc.) to further increase readability.

**Review by loukylor — COMMENTED** [review]
Looks good, just gonna wait for someone to review the actual content 👍

**Review by loukylor — APPROVED** [review]

File: docs/modders/patching.md [hunk 1]
```diff
@@ -143,5 +143,93 @@ Since it would be very annoying having to manually unpatch every single patch in
...
+        GetIl2CppMethodInfoPointerFieldForGeneratedMethod(typeof(UnityEngine.Object).GetMethod("get_name").GetValue(null);
+
+    //storing our patch method in one of the delegate fields
+	_patchDelegate = GetName;
+
+    //getting the IntPtr from _patchDelegate
+	IntPtr delegatePointer = Marshal.GetFunctionPointerForDelegate(_patchDelegate);
+
+    //creating the NativeHook with our target method' IntPtr and patch delegate' IntPtr
+	NativeHook<GetNameDelegate>hook = new NativeHook<GetNameDelegate>(originalMethod, delegatePointer);
```
Comment (Kiokuu) [inline docs/modders/patching.md:191]: s p a c e

File: docs/modders/patching.md [hunk 2]
```diff
@@ -143,5 +143,93 @@ Since it would be very annoying having to manually unpatch every single patch in
...
+the generated assembly if you want the names of every field.
+
+Once we got the ``IntPtr`` for the desired method, the patch method delegate (``_patchDelegate``) is set to hold our patch then 
+we get the IntPtr for it.
+
+A new ``NativeHook`` is made with the generic being our main delegate type (``GetNameDelegate``), the target method' IntPtr 
+and then our patch delegate' IntPtr. We use ``Marshal.GetFunctionPointerForDelegate`` for this.
+
+We call the ``Attach`` method on our newly created ``NativeHook`` (``hook``) to actually hook the target method then we store the hook 
+in that static field we made earlir for it, ``private static NativeHook<GetNameDelegate>Hook``
```
Comment (Kiokuu) [inline docs/modders/patching.md:227]: earlir -\> earlier
private static NativeHook\<GetNameDelegate\>Hook -> private static NativeHook\<GetNameDelegate\> hook

**Pull request merged**